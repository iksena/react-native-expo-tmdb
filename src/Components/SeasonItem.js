import React from 'react';
import {
  Box, Heading, HStack, Image, Text, VStack,
} from 'native-base';
import dayjs from 'dayjs';

import defaultImage from '../../assets/placeholder.jpg';

function SeasonItem(props) {
  const {
    title, date, episodeCount, imagePath, overview,
  } = props;

  return (
    <Box my={2}>
      <HStack>
        <Image
          source={{
            uri: `https://image.tmdb.org/t/p/w200${imagePath}`,
            width: 70,
            height: 100,
          }}
          resizeMode="cover"
          defaultSource={defaultImage}
          fallbackSource={defaultImage}
          alt="poster"
        />
        <VStack marginLeft={2} flex={1}>
          <Heading size="sm">{title}</Heading>
          <Text fontSize="sm">{`${dayjs(date).get('years')} | ${Number(episodeCount)} Episodes`}</Text>
          <Text fontSize="xs">{overview}</Text>
        </VStack>
      </HStack>
    </Box>
  );
}

export default SeasonItem;
