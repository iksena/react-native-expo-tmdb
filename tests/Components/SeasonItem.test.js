import { render } from '../test-utils';
import SeasonItem from '../../src/Components/SeasonItem';

describe('SeasonItem component', () => {
  const props = {
    title: 'Peacemaker',
    date: '2022-04-01',
    episodeCount: '10',
    imagePath: '/.url.jpg',
    overview: 'Peacemaker is not Pissmaker',
  };

  describe('#render', () => {
    it('should render component', () => {
      const { toJSON } = render(<SeasonItem {...props} />);

      expect(toJSON()).toMatchSnapshot();
    });
  });
});
