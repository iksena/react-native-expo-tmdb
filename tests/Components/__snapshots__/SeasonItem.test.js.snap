// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeasonItem component #render should render component 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    style={
      Object {
        "marginBottom": 8,
        "marginTop": 8,
      }
    }
  >
    <View
      dataSet={Object {}}
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Image
        accessibilityLabel="poster"
        alt="poster"
        dataSet={Object {}}
        defaultSource={
          Object {
            "testUri": "../../../assets/placeholder.jpg",
          }
        }
        onError={[Function]}
        resizeMode="cover"
        source={
          Object {
            "height": 100,
            "uri": "https://image.tmdb.org/t/p/w200/.url.jpg",
            "width": 70,
          }
        }
        style={
          Object {
            "maxWidth": "100%",
          }
        }
      />
      <View
        dataSet={Object {}}
        style={Object {}}
      />
      <View
        dataSet={Object {}}
        style={
          Object {
            "flex": 1,
            "marginLeft": 8,
          }
        }
      >
        <Text
          dataSet={Object {}}
          style={
            Object {
              "backgroundColor": undefined,
              "color": "#404040",
              "fontFamily": undefined,
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "700",
              "letterSpacing": 0,
              "lineHeight": 20,
              "textDecorationLine": undefined,
            }
          }
        >
          Peacemaker
        </Text>
        <View
          dataSet={Object {}}
          style={Object {}}
        />
        <Text
          dataSet={Object {}}
          style={
            Object {
              "backgroundColor": undefined,
              "color": "#1f2937",
              "fontFamily": undefined,
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 21,
              "textDecorationLine": undefined,
            }
          }
        >
          2022 | 10 Episodes
        </Text>
        <View
          dataSet={Object {}}
          style={Object {}}
        />
        <Text
          dataSet={Object {}}
          style={
            Object {
              "backgroundColor": undefined,
              "color": "#1f2937",
              "fontFamily": undefined,
              "fontSize": 12,
              "fontStyle": "normal",
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 18,
              "textDecorationLine": undefined,
            }
          }
        >
          Peacemaker is not Pissmaker
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
